src/customers/repository/customer.repository.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@nestjs/common';
+import { Customer } from '../domain/customer.domain';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class CustomerRepository {
+  constructor(
+    @InjectRepository(Customer)
+    private readonly customerRepository: Repository<Customer>,
+  ) {}
+
+  getAllCustomers(): Promise<Customer[]> {
+    return this.customerRepository.find();
+  }
+
+  getCustomerById(id: string): Promise<Customer | null> {
+    return this.customerRepository.findOneBy({ id });
+  }
+
+  async createCustomer(customer: Customer): Promise<Customer> {
```

Comment:
Usually here we don't need to name methods like this: createCustomer, GetAllCustomers etc.
Because when we use them is going to be something like this: `cusomterRepository.createCustomer(...)`
We understand what it does because is called from `customerRepository`
Also this apply for controllers and services as well for basic methods.
**Change this only if you have time but I wanted to let you know about this convention as well.**

Reply:
Here I just wanted to say that you can simplify the naming.
you can use:
- customerRepository.create() instead of cusotmerRepository.createCustomer()
- locationService.update() instead of locationService.updateLocation()
- ...